File: python/aoc01/aoc01.py
#!/usr/bin/env python3

with open("aoc01.dat") as f:
    lines = f.read().splitlines()

# part 1

increases = 0
for i in range(len(lines) - 1):
    if int(lines[i]) < int(lines[i + 1]):
        increases += 1

print(f"part 1 increases: {increases}")

# part 2

increases = 0
for i in range(len(lines) - 3):
    window1 = int(lines[i]) + int(lines[i + 1]) + int(lines[i + 2])
    window2 = int(lines[i + 1]) + int(lines[i + 2]) + int(lines[i + 3])
    if window1 < window2:
        increases += 1

print(f"part 2 increases: {increases}")

File: golang/aoc01/aoc01.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func getWindow(lines []string, index int, length int) int {
	window := 0
	for i := 0; i < length; i++ {
		curVal, err := strconv.Atoi(lines[index+i])
		if err != nil {
			panic(err)
		}
		window += curVal
	}
	return window
}

func main() {

	file, err := os.ReadFile("aoc01.dat")
	if err != nil {
		panic(err)
	}

	lines := strings.Split(string(file), "\n")

	increases := 0
	for i := range lines {
		if i+1 < len(lines) {
			curVal, err := strconv.Atoi(lines[i])
			if err != nil {
				panic(err)
			}

			nexVal, err := strconv.Atoi(lines[i+1])
			if err != nil {
				panic(err)
			}

			if curVal < nexVal {
				increases++
			}
		}
	}

	fmt.Printf("part 1 increases: %v\n", increases)

	increases = 0
	for i := range lines {
		if i+3 < len(lines) {
			win1 := getWindow(lines, i, 3)
			win2 := getWindow(lines, i+1, 3)
			if win1 < win2 {
				increases++
			}
		}
	}

	fmt.Printf("part 2 increases: %v\n", increases)
}
